processor/metricstransformprocessor/metrics_transform_processor_helper.go

```
@@ -0,0 +1,375 @@
+				Timestamp: timestamp,
+				Value:     doublePoint,
+			})
+		} else {
+			for _, p := range distPoints {
+				newPoints = append(newPoints, &metricspb.Point{
+					Timestamp: timestamp,
+					Value:     p,
+				})
+			}
+		}
+	}
+	return newPoints
+}
+
+// compute attempts to merge points into one point based on the provided aggrType and dataType
+// Returns an int value, a double value and a slice of distribution values, and only one of these would contain values based on the dataType.
+func (mtp *metricsTransformProcessor) compute(points []*metricspb.Point, aggrType AggregationType, dataType metricspb.MetricDescriptor_Type) (*metricspb.Point_Int64Value, *metricspb.Point_DoubleValue, []*metricspb.Point_DistributionValue) {
+	intVal := int64(0)
+	doubleVal := float64(0)
+	distVals := make([]*metricspb.Point_DistributionValue, 0)
+	switch dataType {
+	case metricspb.MetricDescriptor_GAUGE_INT64, metricspb.MetricDescriptor_CUMULATIVE_INT64:
+		for _, p := range points {
+			if aggrType == Sum || aggrType == Average {
+				intVal += p.GetInt64Value()
+			} else if aggrType == Max {
+				if p.GetInt64Value() <= intVal {
+					continue
+				}
+				intVal = p.GetInt64Value()
+			}
+
+		}
+		if aggrType == Average {
+			intVal /= int64(len(points))
+		}
+		return &metricspb.Point_Int64Value{Int64Value: intVal}, nil, nil
+	case metricspb.MetricDescriptor_GAUGE_DOUBLE, metricspb.MetricDescriptor_CUMULATIVE_DOUBLE:
+		for _, p := range points {
+			if aggrType == Sum || aggrType == Average {
+				doubleVal += p.GetDoubleValue()
+			} else if aggrType == Max {
+				if p.GetDoubleValue() <= doubleVal {
+					continue
+				}
+				doubleVal = p.GetDoubleValue()
+			}
+
+		}
+		if aggrType == Average {
+			doubleVal /= float64(len(points))
+		}
+		return nil, &metricspb.Point_DoubleValue{DoubleValue: doubleVal}, nil
+	case metricspb.MetricDescriptor_GAUGE_DISTRIBUTION, metricspb.MetricDescriptor_CUMULATIVE_DISTRIBUTION:
+		if aggrType != Sum {
+			break
```

Is this a TODO? Maybe this should raise an error?

---

This is not a TODO item, because based on the current design it would only make sense for histograms to perform summation for aggregation. Do you think I should log errors here or make it explicit in the documentation?

---

I think this should log a warning at least, because this would mean someone has set up an invalid configuration.